### PR TITLE
Implement loading of room layouts from edit files

### PR DESCRIPTION
**Pull request by nomadicwriter**
Adds and parses room_template.txt edit file.  Generates special rooms from layouts with variable door/object/monster placement.

### CODE REVIEW

**Comment by magnate** [comment]
Fantastic! I'll wait for d_m's comments before merging, in case it clashes with his broader aims for dungeon generation, but it looks great. Well done.

File: src/generate.c [hunk 1]
```diff
@@ -1866,6 +1889,165 @@ static bool build_pit(struct cave *c, int y0, int x0)
 	return TRUE;
 }
 
+/**
+ * Build a room template from its string representation.
+ */
+
+
+static void build_room_template(struct cave *c, int y0, int x0, int ymax, int xmax, int doors, const char *data)
```
Comment (takkaria) [inline src/generate.c:1897]: The indentation in this function is really uneven - looks like a mixture of tabs and spaces.  Please could you follow the http://trac.rephial.org/wiki/CodingGuidelines?  Thanks!

Comment (nomadicwriter) [inline src/generate.c:1897]: Ack, sorry.  Still figuring out CodeBlocks and its automatic line spacing.  Will fix!

**Comment by magnate** [comment]
Merged - thank you

**Pull request merged**